Lib/fontTools/otlLib/builder.py

```diff
@@ -544,23 +1111,53 @@ def buildAttachPoint(points):
 
 
 def buildCaretValueForCoord(coord):
-    """500 --> otTables.CaretValue, format 1"""
+    # 500 --> otTables.CaretValue, format 1
     self = ot.CaretValue()
     self.Format = 1
     self.Coordinate = coord
     return self
 
 
 def buildCaretValueForPoint(point):
-    """4 --> otTables.CaretValue, format 2"""
+    # 4 --> otTables.CaretValue, format 2
     self = ot.CaretValue()
     self.Format = 2
     self.CaretValuePoint = point
     return self
 
 
 def buildLigCaretList(coords, points, glyphMap):
-    """{"f_f_i":[300,600]}, {"c_t":[28]} --> otTables.LigCaretList, or None"""
+    """Builds a ligature caret list table.
+
+    Ligatures appear as a single glyph representing multiple characters; however
+    when, for example, editing text containing a ``f_i`` ligature, the user may
+    want to place the cursor between the ``f`` and the ``i``. The ligature caret
+    list in the GDEF table specifies the position to display the "caret" (the
+    character insertion indicator, typically a flashing vertical bar) "inside"
+    the ligature to represent an insertion point. The insertion positions may
+    be specified either by coordinate or by contour point.
+
+    Example::
+
+        coords = {
+            "f_f_i": [300, 600] # f|fi cursor at 300 units, ff|i cursor at 600.
+        }
+        points = {
+            "c_t": [28] # c|t cursor appears at co-ordinate of contour point 28.
+        }
+        ligcaretlist = buildLigCaretList(coords, points, font.getReverseGlyphMap())
+
+    Args:
+        coords: A mapping between glyph names and a list of X-coordinates for
```

Comment:
The spec says “Caret coordinates are either X or Y values, depending upon the text direction.” Does this function only support X values?